frontend/src/components/templates/Form/HideSectionsContainer.tsx

```diff
@@ -0,0 +1,110 @@
+import React, {
+  Children,
+  createContext,
+  FunctionComponent,
+  useEffect,
+  useState,
+} from 'react';
+import Button from '../../atoms/hyperTexts/Button';
+import Loader from '../../atoms/Loader';
+
+export type HideSectionsContextType = {
+  setReadyForNextSteps: (v: boolean) => void;
+};
+
+export const HideSectionsContext = createContext<HideSectionsContextType>({
+  setReadyForNextSteps: () => null,
+});
+
+type Props = {
+  SubmissionPanel: FunctionComponent;
+};
+
+export const HideSectionsContainer: FunctionComponent<Props> = ({
+  SubmissionPanel,
+  children,
+}) => {
+  const [showOnlyFirstStep, setShowOnlyFirstStep] = useState(false);
+  const [isShowNextStepButtonDisabled, setIsShowNextStepButtonDisabled] =
+    useState(false);
+  const [areNextStepsLoading, setAreNextStepLoading] = useState(false);
+  const [readyForNextSteps, setReadyForNextSteps] = useState(true);
+
+  useEffect(() => {
+    if (showOnlyFirstStep) {
+      setIsShowNextStepButtonDisabled(true);
+    }
+  }, [setIsShowNextStepButtonDisabled, showOnlyFirstStep]);
+
+  const showNextStep = () => {
```

Comment:
👍 this is the right way to do it, an handler that have the logic and mutate multiple state hooks